### PR TITLE
Added add/mul for nested dense [B, *, D], [B, 1, D] case (CUDA-only)

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #88289

### CODE REVIEW

File: aten/src/ATen/native/nested/NestedTensorBinaryOps.cpp [hunk 1]
```diff
@@ -207,5 +242,16 @@ Tensor& fill_nested_(Tensor& self, const Tensor& value) {
...
 }
 
+void _nested_op_dense_esuhm_cpu(Tensor& result, const Tensor& self, const Tensor& other, const NESTED_DENSE_OP& op) {
+  // TODO: implement CPU kernel
+  TORCH_CHECK(false);
+}
+
+REGISTER_ARCH_DISPATCH(nested_dense_elementwise_stub, DEFAULT, &_nested_op_dense_esuhm_cpu);
+REGISTER_AVX512_DISPATCH(nested_dense_elementwise_stub, &_nested_op_dense_esuhm_cpu);
+REGISTER_AVX2_DISPATCH(nested_dense_elementwise_stub, &_nested_op_dense_esuhm_cpu);
```
Comment (cpuhrsch) [inline aten/src/ATen/native/nested/NestedTensorBinaryOps.cpp:252]: nit: We don't need this for now. We can keep the CPU code in this file.

Comment (mikaylagawarecki) [reply aten/src/ATen/native/nested/NestedTensorBinaryOps.cpp:252]: I wasn't able to get it to compile when I only registered the cuda dispatch

Comment (cpuhrsch) [reply aten/src/ATen/native/nested/NestedTensorBinaryOps.cpp:252]: You can try `REGISTER_NO_CPU_DISPATCH `https://github.com/pytorch/pytorch/blob/6faa6c68e8b76fb68f3a2b2783685102d0e87c00/aten/src/ATen/native/DispatchStub.h#L265

**Comment by cpuhrsch** [comment]
Current test failures seem unrelated

**Review by cpuhrsch — APPROVED** [review]
Generally looks good (see last nit). Also we'll need to mark this CUDA kernel for iterative improvements once we have end to end benchmarks.

**Pull request closed without merging**